src/process_utility.c

```diff
@@ -328,19 +326,26 @@ process_drop_table(DropStmt *stmt)
 			ht = hypertable_cache_get_entry(hcache, relid);
 			if (NULL != ht)
 			{
+				CatalogSecurityContext sec_ctx;
+
 				if (list_length(stmt->objects) != 1)
 					elog(ERROR, "Cannot drop a hypertable along with other objects");
 
+				catalog_become_owner(catalog_get(), &sec_ctx);
```

Comment:
Now the SQL upcalls require the precondition of being called while being the catalog owner, although they are accessible (and callable) by anyone (as opposed to a C function that cannot be called easily). At the very least, we should probably document that these functions only work if called as the catalog owner (in case a user calls, e.g., drop_hypertable manually, or someone changing the SQL code calls that function). Maybe also add a user check in the functions to ensure that they are called with the right user.

Reply:
So 2 points:

Reply:
So 2 points:
1) These functions don't have a precondition of being run as any user. They will just fail with a permission error if the user doesn't have the right permissions. This is both safe and the right thing to do.
2) Functions in _timescaledb_internal are not for user-consumption. I don't see a need to document any of these extensively. Although I can if you insist.